addons/binding/org.openhab.binding.dwdunwetter/src/main/java/org/openhab/binding/dwdunwetter/internal/data/DwdWarningCache.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2010-2019 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dwdunwetter.internal.data;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.openhab.binding.dwdunwetter.internal.DwdUnwetterBindingConstants;
+
+/**
+ * Cache of Warnings to update the {@link DwdUnwetterBindingConstants#CHANNEL_UPDATED} if a new warning is sent to a
+ * channel.
+ *
+ * @author Martin Koehler - Initial contribution
+ */
+public class DwdWarningCache {
```

Comment:
That doesn't seem to fit my requirements. I just want to prevent a memory leak - I don't need a "reload" action. I just want to remove the entries after the warning has expired,